src/jit/hwintrinsiclistxarch.h

```diff
@@ -50,7 +53,10 @@ HARDWARE_INTRINSIC(Base_Vector256_AsSingle,                         "AsSingle",
 HARDWARE_INTRINSIC(Base_Vector256_AsUInt16,                         "AsUInt16",                                    Base,          -1,              32,           0,     {INS_invalid,           INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},           HW_Category_Helper,                 HW_Flag_NoContainment|HW_Flag_NoRMWSemantics)
 HARDWARE_INTRINSIC(Base_Vector256_AsUInt32,                         "AsUInt32",                                    Base,          -1,              32,           0,     {INS_invalid,           INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},           HW_Category_Helper,                 HW_Flag_NoContainment|HW_Flag_NoRMWSemantics)
 HARDWARE_INTRINSIC(Base_Vector256_AsUInt64,                         "AsUInt64",                                    Base,          -1,              32,           0,     {INS_invalid,           INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},           HW_Category_Helper,                 HW_Flag_NoContainment|HW_Flag_NoRMWSemantics)
-HARDWARE_INTRINSIC(Base_Vector256_Zero,                             "get_Zero",                                    Base,          -1,              32,           0,     {INS_invalid,           INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},           HW_Category_Helper,                 HW_Flag_NoContainment|HW_Flag_NoRMWSemantics)
+HARDWARE_INTRINSIC(Base_Vector256_CreateScalarUnsafe,               "CreateScalarUnsafe",                          Base,          -1,              32,           1,     {INS_mov_i2xmm,         INS_mov_i2xmm,      INS_mov_i2xmm,      INS_mov_i2xmm,      INS_mov_i2xmm,      INS_mov_i2xmm,      INS_mov_i2xmm,      INS_mov_i2xmm,      INS_movss,          INS_movsdsse2},         HW_Category_Helper,                 HW_Flag_NoContainment|HW_Flag_NoRMWSemantics)
+HARDWARE_INTRINSIC(Base_Vector256_GetLower,                         "GetLower",                                    Base,          -1,              32,           1,     {INS_movdqu,            INS_movdqu,         INS_movdqu,         INS_movdqu,         INS_movdqu,         INS_movdqu,         INS_movdqu,         INS_movdqu,         INS_movups,         INS_movupd},            HW_Category_Helper,                 HW_Flag_NoContainment|HW_Flag_NoRMWSemantics)
```

Comment:
No, only for reg-reg moves. The purpose of these intrinsics (`ToScalar`, `GetLower`, etc) is to reuse the existing registers as other "types", so I think it might not worthy to handle them with containment.

Reply:
For example, `v = ToScalar(floatArry[i])`, finially it should generate one instruction
```asm
movss xmm0, [addr]
```
It does not matter `movss` is from `floatArry[i]` or `ToScalar`. So, not handling these guys in containment is better.

Reply:
I think that would only be true if we could guarantee a reg-reg move (which wont always be possible/feasible, and may not be optimal, depending on where the input is currently).

Reply:
Handling `GetLower` is impossible and useless. For example, the optimal codegen of `Add(v1, LoadVector256(addr).GetLower()` should be
```asm
vmovups ymm2, [addr]
;;; GetLower has no codegen
vaddps xmm1, xmm1, xmm2
```

Reply:
> For example, the optimal codegen of Add(v1, LoadVector256(addr).GetLower()) should be
> ```asm
> vmovups ymm2, [addr]
> vaddps xmm1, xmm1, xmm2
> ```

I would say it would be better if the JIT just handled that as:
```asm
vaddps xmm1, xmm1 [addr]
```

There is also the case where you have an existing local `Vector256<T>` that was stack-spilled (and similar analogies for the other helper-intrinsics here)

Reply:
> an existing local Vector256<T> that was stack-spilled 

Yes, I think this is the only common case. But I am not sure it worthy or not.

Reply:
>>an existing local Vector256 that was stack-spilled

> Yes, I think this is the only common case. But I am not sure it worthy or not.

And I think `GetLower` is the only one. `CreateScalarUnsafe`, `ToVector256`, and other "widening" operations should not generate load instructions.

Reply:
> I would say it would be better if the JIT just handled that as:

> vaddps xmm1, xmm1 [addr]

This codegen needs to folding `GetLower` in `Add`.

Reply:
> And I think `GetLower` is the only one.

I wouldn't agree here. The others also have the chance for the input to have been stack-spilled and for which we can generate more efficient code by handling containment.

For `ToVector256`, we would emit the same `movups`, `movupd`, or `movdqu` instruction and the upper-bits would be zeroed.
For `CreateScalarUnsafe`, we would also still emit the appropriate `movss` or `movsd` instruction, loading from memory.

For all of these, the input has a decent chance of being on the stack (either from stack-spillage or because it was passed on the stack as an input argument from the caller).

Reply:
> For ToVector256, we would emit the same movups, movupd, or movdqu instruction and the upper-bits would be zeroed.
For CreateScalarUnsafe, we would also still emit the appropriate movss or movsd instruction, loading from memory.

That would be exactly same with generating the code from their argument instead if themselves.

For optimal codegen like `vaddps xmm1, xmm1 [addr]` we need to fold these intrinsic (like `GetLower`) nodes away in some phases. That is not related not this topic.

But anyway, currently, I suggest only generating `movaps reg, reg` for these intrinsic if necessary. `movaps` has shorter code size, unified ISA support, and zero-latency performance.

Reply:
I've updated this to use `movaps` for `reg, reg` moves. I am however, also updating these to support containment as there are some pretty trivial cases (even when just trying to implement the other helper-intrinsics) where it helps codegen.

I think we could also support doing more efficient things when we encounter some of these helper intrinsics, but that can be separate handling in addition to the containment support (which is trivial to add anyways).

Reply:
> I am however, also updating these to support containment as there are some pretty trivial cases (even when just trying to implement the other helper-intrinsics) where it helps codegen.

Perhaps updating in a separate PR is better to see the codegen diff?

Reply:
I just posted the diffs for both with and without the containment support.